src/core/port-range-access.h

```diff
@@ -0,0 +1,37 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+#pragma once
+
+/***
+  This file is part of systemd.
+
+  Copyright 2016 Daniel Mack
+  Copyright 2017 Intel Corporation.
+
+  systemd is free software; you can redistribute it and/or modify it
+  under the terms of the GNU Lesser General Public License as published by
+  the Free Software Foundation; either version 2.1 of the License, or
+  (at your option) any later version.
+
+  systemd is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with systemd; If not, see <http://www.gnu.org/licenses/>.
+***/
+
+#include "list.h"
+
+typedef struct PortRangeAccessItem PortRangeAccessItem;
+
+struct PortRangeAccessItem {
+        unsigned char prefixlen;
+        uint16_t port;
+        uint16_t protocol;
```

Comment:
u8 should be right, because the protocol field is only eight bits. My initial version had the value as u8, but kernels < 4.14 fail then with bpf stack alignment problems. A cleaner solution would have a first a zeroed padding byte and then the protocol as u8.

Reply:
i'd still make it an "int" AFAICS we use an "int" for it at most other places, and the majority of APIs appears to use int, too